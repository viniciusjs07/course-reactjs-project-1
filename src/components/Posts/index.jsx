import { PostCard } from '../PostCard';
import PropTypes from 'prop-types';
import './styles.css';

export const Posts = ({ posts }) => {
  return (
    <div className="posts">
      {posts?.map((post) => (
        <PostCard post={post} key={post.id}></PostCard>
      ))}
    </div>
  );
};

Posts.propTypes = {
  posts: PropTypes.arrayOf(PropTypes.object).isRequired,
};
export default Posts;
