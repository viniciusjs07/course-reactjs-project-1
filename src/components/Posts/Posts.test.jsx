import { render, screen } from '@testing-library/react';

import Posts from '.';

const posts = [
  {
    userId: 1,
    id: 1,
    title: 'Title 1',
    body: 'Body 1',
    cover: 'img/img1.png',
  },
  {
    userId: 2,
    id: 2,
    title: 'Title 2',
    body: 'Body 2',
    cover: 'img/img2.png',
  },
  {
    userId: 3,
    id: 3,
    title: 'Title 3',
    body: 'Body 3',
    cover: 'img/img3.png',
  },
];

describe('<Posts/>', () => {
  console.log('posts ', posts);
  it('Should render Posts', () => {
    render(<Posts posts={posts} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByAltText(/title/i)).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'img/img3.png');
  });

  it('Should match snapshot', () => {
    const { container } = render(<Posts posts={posts} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
