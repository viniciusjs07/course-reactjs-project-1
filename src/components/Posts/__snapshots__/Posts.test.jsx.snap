// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> Should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 1
      </h2>
      <p>
        Body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 2
      </h2>
      <p>
        Body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 3
      </h2>
      <p>
        Body 3
      </p>
    </div>
  </div>
</div>
`;
